src/pykeen/pipeline.py

```diff
@@ -250,16 +250,142 @@ def title(self) -> Optional[str]:  # noqa:D401
         """The title of the experiment."""
         return self.metadata.get('title')
 
-    def plot_losses(self):
+    def plot_losses(self, ax=None):
         """Plot the losses per epoch."""
-        import matplotlib.pyplot as plt
+        if ax is None:
+            import matplotlib.pyplot as plt
+            ax = plt.gca()
+
+        import seaborn as sns
+        sns.set_style('darkgrid')
+
+        rv = sns.lineplot(x=range(len(self.losses)), y=self.losses, ax=ax)
+
+        loss_name = normalize_string(self.model.loss.__class__.__name__, suffix=_LOSS_SUFFIX)
+        ax.set_ylabel(f'{loss_name} Loss')
+        ax.set_xlabel('Epoch')
+        ax.set_title(self.title if self.title is not None else 'Losses Plot')
+        return rv
+
+    def plot_er(  # noqa: C901
+        self,
+        model: Optional[str] = None,
+        margin: float = 0.4,
+        ax=None,
+        entities: Optional[Set[str]] = None,
+        relations: Optional[Set[str]] = None,
+        apply_limits: bool = True,
+        plot_entities: bool = True,
+        plot_relations: bool = True,
+        annotation_x_offset: float = 0.02,
+        annotation_y_offset: float = 0.03,
+        **kwargs,
+    ):
+        """Plot the reduced entities and relation vectors in 2D."""
+        if not plot_entities and not plot_relations:
+            raise ValueError
+
+        if model is None:
+            model = 'PCA'
+        if model.upper() == 'PCA':
+            from sklearn.decomposition import PCA as Reducer  # noqa:N811
+        elif model.upper() == 'KPCA':
+            kwargs.setdefault('kernel', 'rbf')
+            from sklearn.decomposition import KernelPCA as Reducer
+        elif model.upper() == 'GRP':
+            from sklearn.random_projection import GaussianRandomProjection as Reducer
+        elif model.upper() == 'SRP':
+            from sklearn.random_projection import SparseRandomProjection as Reducer
```

Comment:
@cthoyt could you also add the methods from sklearn's [manifold learning section](https://scikit-learn.org/stable/auto_examples/manifold/plot_compare_methods.html#sphx-glr-auto-examples-manifold-plot-compare-methods-py)? At least t-SNE is quite popular to visualize embeddings (although there are some pitfalls/challenges when interpreting it, cf. e.g. https://distill.pub/2016/misread-tsne/ )

Reply:
However, for these non-linear methods, the relation plotting might not be meaningful anymore.

So maybe we extract an additional methods `plot_e`?` Or emit a warning when we try to plot relations? Also the interpretation of the relation vector as offset, is unique to the trans* family of methods.

Reply:
yeah, I guess the above comment to make this non-default would be good enough.

Reply:
Addressed additional reducers in e4432aa4

Made relation plot non-default in 567eb0bf